core/nonce.go

```diff
@@ -21,19 +21,30 @@ const StateKeyNonce byte = 42 // hex: 2a
 const (
 	doublingMemoryCoef    = 2
 	LenTimeInMilliseconds = 13
-	// defaultNonceTTL is time in seconds for nonce. If attempting to execute a transaction in a batch
+	// defaultNonceTTL is time for nonce. If attempting to execute a transaction in a batch
 	// that is older than the maximum nonce (at the current moment) by more than NonceTTL,
 	// we will not execute it and return an error.
-	defaultNonceTTL = 50
+	defaultNonceTTL               = 50 * time.Second
+	defaultNonceTTLForCreateCCTTo = 3 * time.Hour
 )
 
-func checkNonce(
+type Nonce struct {
+	nonceTTL   time.Duration
+	guardF     func(uint64) error
+	transformF func(uint64) uint64
+}
+
+func (n *Nonce) checkNonce(
```

Comment:
nonce.checkNonce should be nonce.check(value)

Reply:
fix